cli/test/grpc.test.ts

```diff
@@ -0,0 +1,145 @@
+import { rmSync, mkdirSync, existsSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+import { tmpdir } from 'node:os';
+import { Command } from 'commander';
+import { describe, test, expect } from 'vitest';
+import { createPromiseClient, createRouterTransport } from '@connectrpc/connect';
+import { PlatformService } from '@wundergraph/cosmo-connect/dist/platform/v1/platform_connect';
+import GenerateCommand from '../src/commands/grpc/commands/generate.js';
+import { Client } from '../src/core/client/client.js';
+
+export const mockPlatformTransport = () =>
+  createRouterTransport(({ service }) => {
+    service(PlatformService, {});
+  });
+
+describe('gRPC Generate Command', () => {
+  test('should generate proto and mapping files', async () => {
+    const client: Client = {
+      platform: createPromiseClient(PlatformService, mockPlatformTransport()),
+    };
+
+    const program = new Command();
+    program.addCommand(GenerateCommand({ client }));
+
+    const tmpDir = join(tmpdir(), `grpc-test-${Date.now()}`);
+    mkdirSync(tmpDir, { recursive: true });
```

Comment:
There are also cleanup functions that always run after a test regardless of its outcome.